### DISCUSSION TITLE
MemoryError when running ReEDS

### DISCUSSION THREAD

**Discussion by yangqiu91** [NREL/ReEDS-2.0#51] [Q&A: Execution] [answered]
Hi ReEDS team,

I am getting started with ReEDS by running the scenarios in the cases_standardscenarios.csv file. When I was running one case (Mid_Case) under the sequential solve structure, I had this MemoryError (Screenshot below), which means my computer run out of memory. 
![image](https://user-images.githubusercontent.com/36561597/130002994-f1cd19c8-e93f-4ffd-83ee-8cd2e35d572a.png)

My computer has installed RAM of 16 GB, and I also installed a 64-bit version python. I looked at the hardware requirement to run ReEDS, it says the **RAM requirement for running one instance (sequential solve structure) is 4-7 GB**, so I guess my computer should have enough RAM memory to run at least 1 instance/case at a time. Right? Please let me know if you have any thoughts to address this. Do I need to increase the installed RAM of my computer? 

Yang

**Comment by vcelloho** [comment, marked as answer]
The guidance on memory depends on the options selected and the available memory at the time of running the model. 16 GB should be sufficient, but you'll need to check what usage your computer is using for other programs. Web tabs can be very consumptive. When next running I recommend checking the memory tab in task manager it will be helpful in diagnosing the issue.

**Reply by yangqiu91** [reply]
Thank you for your feedback.
Yes, I closed all unnecessary programs when I ran the model. I have about 10.9 GB RAM in the beginning. The memory usage increases as I start to run ReEDS, however when the MemoryError occurs, there is still plenty of available memory (following screenshot). Is there any other possible reason for this error? 
![image](https://user-images.githubusercontent.com/36561597/130104417-75a52e66-55bb-47a0-af8a-2d334e05fdb3.png)

**Reply by yangqiu91** [reply]
For some reason, when I run ReEDS, it, by default, executes scripts using the Python under this directory (C:\Python27\ArcGIS10.7), which is a 32-bit Python. 

Then, when I went to the ReEDS directory in my computer, and check the Python version, it says 64-bit. So **what I did was to add "Python" before "runbatch.py", and the problem was solved, and the model ran successfully (Screenshot below).** I still do not fully understand the reason behind this, but obviously the problem was caused by the version of python (32-bit vs 64 bit). I just want to provide my solution here in case people have a similar issue in the future. 
![image](https://user-images.githubusercontent.com/36561597/130262659-9aa837c4-a5d4-4322-a990-6708f9087358.png)

**Reply by vcelloho** [reply]
I see, the reason for this is that Python27 may be on the path variable for Python.
My recommendation is to ensure you have the directory containing the 64 bit version of Python in the Path as part of your environmental variables, and at least if it is the case remove the ArcGIS version of Python from your path variables.

You can access your environmental variables using the following tutorial.

https://www.architectryan.com/2018/08/31/how-to-change-environment-variables-on-windows-10/
